ckan/tests/controllers/test_package.py

```diff
@@ -1503,9 +1506,15 @@ def test_search_plugin_hooks(self, app):
     def test_search_page_request(self, app):
         """Requesting package search page returns list of datasets."""
 
-        factories.Dataset(name="dataset-one", title="Dataset One")
-        factories.Dataset(name="dataset-two", title="Dataset Two")
-        factories.Dataset(name="dataset-three", title="Dataset Three")
+        factories.Dataset(
+            name="dataset-one", title="Dataset One", type="dataset"
+        )
+        factories.Dataset(
+            name="dataset-two", title="Dataset Two", type="dataset"
+        )
+        factories.Dataset(
+            name="dataset-three", title="Dataset Three", type="dataset"
+        )
```

Comment:
In order to get the tests to pass after I made this change, I have to explicitly set type="dataset", on every dataset we create in this test suite. Would it be better to just move this to the base implementation of `factories.Dataset`?

Reply:
Yes please, add a `type = dataset` property on the Dataset factory